src/resmom/linux/alps.c

```diff
@@ -5102,8 +5262,19 @@ alps_create_reserve_request(job *pjob, basil_request_reserve_t **req)
 
 		p->depth = ns->depth;
 		p->nppn = width = ns->width;
-		p->rsvn_mode = (ns->share == isexcl) ?
-		basil_rsvn_mode_exclusive : basil_rsvn_mode_shared;
+
+		/*
+		 * If the user requested place=excl then we need to pass
+		 * that information into the ALPS reservation.
+		 */
+		p->rsvn_mode = basil_rsvn_mode_shared; /* initialize it */
+		if (rpv == rlplace_excl) {
+			/*
+			 * The user asked for the node exclusively.
+			 * Set it in the ALPS reservation
+			 */
+			p->rsvn_mode = basil_rsvn_mode_exclusive;
+		}
```

Comment:
What was wrong with the tertiary operator? The conditional has changed, but it saves space.

Why did the conditional change?